### PR TITLE
[Issue #1362] Sendy Update

### LINKED ISSUE #1362: [Task]: Move API Handlers to newer app router "Route Handlers"
### Summary

Both the health check and Sendy API route are using the page router "API Handler".

We should move both to Route Handlers : https://nextjs.org/docs/app/building-your-application/routing/route-handlers

### Acceptance criteria

- Sendy API route working 
- Health check API working
- Tests passing

### PR DESCRIPTION
**Pull request by acouch**
## Summary
Fixes #1362

## Changes proposed
This builds on https://github.com/HHS/simpler-grants-gov/pull/2428 by adding tests. See that PR for details.

### CODE REVIEW

**Review by emilycnava — APPROVED** [review]
Pair reviewed with Aaron, discussed changes between original draft PR and this one and it all LGTM!

**Review by doug-s-nava — COMMENTED** [review]
Everything that's here looks good but curious why the e2e test was deleted. Will go through some manual testing now

**Review by doug-s-nava — COMMENTED** [review]
* when running an npm install on the branch I get a small diff in the package-lock removing `"@types/jest": "^29.5.13",`. Seems like either something we should figure out or commit

* I get a hydration error on the subscribe page when I load it up, see screenshot

* I get a "failed to susbscrbe, due to a server error. Please try again later." message on the frontend when trying to subscribe. On the Next backend I see `Error subscribing user: Failed to parse URL from /subscribe`

Screenshot 2024-10-28 at 9 37 20 AM

File: frontend/src/app/[locale]/subscribe/actions.ts
Comment (acouch) [reply frontend/src/app/[locale]/subscribe/actions.ts:43]: Looks like it is [discouraged by the maintainer](https://github.com/amannn/next-intl/discussions/1134#discussioncomment-9796463) but there is a [better way](https://github.com/amannn/next-intl/discussions/1134#discussioncomment-10940240). Let's try that.

File: frontend/src/app/[locale]/subscribe/actions.ts
Comment (acouch) [reply frontend/src/app/[locale]/subscribe/actions.ts:98]: Seems overkill for me to create a provider class for an API that is called once. If we end up with more than one newsletter form then it would make more sense to me.

File: frontend/src/i18n/messages/en/index.ts
Comment (acouch) [reply frontend/src/i18n/messages/en/index.ts:351]: ❓

File: frontend/tests/pages/subscribe/SubscriptionForm.test.tsx
Comment (acouch) [reply frontend/tests/pages/subscribe/SubscriptionForm.test.tsx:37]: Some more details: https://github.com/HHS/simpler-grants-gov/pull/2428#issuecomment-2422833098

Clicking buttons seems weird to me for unit tests but I guess that is react for you. The issue in Next I linked to suggests that button clicks should be supported in React 19 but would take that with a grain of 🧂 . Short term it looks like we can't click buttons in jest, but long term we might be able to. Not sure where the best place to document this is as it is a moving target.

**Comment by acouch** [comment]
@doug-s-nava 

> I get a "failed to susbscrbe, due to a server error. Please try again later." message on the frontend when trying to subscribe. On the Next backend I see Error subscribing user: Failed to parse URL from /subscribe

I think this might be because you don't have the sendy credentials? I updated the docs to reflect needing that: https://github.com/HHS/simpler-grants-gov/pull/2530/commits/94e62a847567d9ebc9fbbb3f96e653163b7648de

> Everything that's here looks good but curious why the e2e test was deleted. Will go through some manual testing now

They don't work currently with server actions. Created a ticket to fix that: https://github.com/HHS/simpler-grants-gov/issues/2537

File: frontend/src/i18n/messages/en/index.ts
Comment (doug-s-nava) [reply frontend/src/i18n/messages/en/index.ts:351]: Just curious where the impetus for changing "<email_address/> is already subscribed. If you’re not seeing our emails, check your spam folder and add no-reply@grants.gov to your contacts, address book, or safe senders list. If you continue to not receive our emails, contact <email>simpler@grants.gov</email>." to "This email address has already been subscribed." is coming from exactly since I didn't see it in any requirements. Very possible I just missed that req somewhere

File: frontend/src/app/[locale]/subscribe/actions.ts
Comment (doug-s-nava) [reply frontend/src/app/[locale]/subscribe/actions.ts:98]: I would tend to agree there. However, it seems like the system we currently have is kind of designed to be built for one off requests. Do we want to revisit that pattern?

File: frontend/src/app/[locale]/subscribe/actions.ts
Comment (doug-s-nava) [reply frontend/src/app/[locale]/subscribe/actions.ts:43]: Very weird. It seems like they're saying they discourage passing the function around _because_ the typing is cumbersome, rather than that they specifically left the typing cumbersome in order to discourage passing the function around for some more thought out reason.

I can see not wanting to pass this function around for one particular reason: if you pass `t` from a server component to a client component you end up potentially closing over the entire messages object, and shipping the entire text content of your site to the client unnecessarily. I'm not exactly sure how next-intl works or if that is expected as the cost of doing business but it seems potentially, and this is a very scientific word here, slops.

File: frontend/src/i18n/messages/en/index.ts
Comment (acouch) [reply frontend/src/i18n/messages/en/index.ts:351]: Sorry, I should have been more verbose. Not sure where that came from myself. @emilycnava ?

**Review by doug-s-nava — COMMENTED** [review]
Local test with the correct envs checks out. Holding on approval for clarity around the "already subscribed" messaging change but the rest all LGTM

File: frontend/src/app/[locale]/subscribe/actions.ts
Comment (acouch) [reply frontend/src/app/[locale]/subscribe/actions.ts:98]: Let's discuss! I added to the agenda for Thurs.

File: frontend/src/i18n/messages/en/index.ts
Comment (emilycnava) [reply frontend/src/i18n/messages/en/index.ts:351]: It wasn't a req, just a choice I made due to the positioning of the error message being different and long messages looking weird. Also the email var insertion code was broken with the old approach.

Old:
Screenshot 2024-10-28 at 3 07 53 PM

New:
Screenshot 2024-10-28 at 3 11 41 PM

**Comment by emilycnava** [comment]
@doug-s-nava Answered your Q r.e. already_subscribed msg above :)

File: frontend/src/i18n/messages/en/index.ts
Comment (doug-s-nava) [reply frontend/src/i18n/messages/en/index.ts:351]: I'm ok with this since it's fixing a bug with the formatting, but we should probably get sign off from design / content on the final text

File: frontend/src/app/[locale]/subscribe/actions.ts
Comment (doug-s-nava) [inline frontend/src/app/[locale]/subscribe/actions.ts:38]: I wonder if the real solution here is to just pass along the strings that the function needs rather than passing down the entire "t" function. That would likely mean passing a lot of strings, though, or basically partially recreating the messaging object in order to pass through just what we need. In any case I'm still worried about the entire message object getting sent down to the client, but not enough to continue blocking this PR, especially since I don't see an immediately ideal solution otherwise.

File: frontend/src/app/[locale]/subscribe/actions.ts
Comment (doug-s-nava) [inline frontend/src/app/[locale]/subscribe/actions.ts:21]: if we do 
```suggestion
  const t = await getTranslations("Subscribe");
```
do we only close over that part of the messaging object? if so that could be a nice middle ground here

**Review by doug-s-nava — APPROVED** [review]
a few qualms but nothing to block this from going out. Really good important stuff!


✅ ✅ ✅ ✅

File: frontend/src/app/[locale]/subscribe/actions.ts
Comment (acouch) [reply frontend/src/app/[locale]/subscribe/actions.ts:21]: `await getTranslations("Subscribe");` caused a linting error.

**Pull request merged**
